langchain4j-vertex-ai-gemini/src/main/java/dev/langchain4j/model/vertexai/VertexAiGeminiChatModel.java

```diff
@@ -101,16 +99,51 @@ public VertexAiGeminiChatModel(GenerativeModel generativeModel,
 
     @Override
     public Response<AiMessage> generate(List<ChatMessage> messages) {
-
         List<Content> contents = ContentsMapper.map(messages);
-
         GenerateContentResponse response = withRetry(() -> generativeModel.generateContent(contents, generationConfig), maxRetries);
 
-        return Response.from(
+        Content content = ResponseHandler.getContent(response);
+        Optional<Part> containsFunctionCall = content.getPartsList().stream().filter(Part::hasFunctionCall).findFirst();
```

Comment:
minor: in theory, there should be no funciton calls when using `generate(List<ChatMessage>)` method because there are no tool specs to start with. But user could use both `generate(List<ChatMessage>)` and `generate(List<ChatMessage>, List<ToolSpecification>)` and `generate(List<ChatMessage>, List<ToolSpecification>)` calls `generate(List<ChatMessage>)`. It is just a bit unobvious from the current implementation, maybe worth adding a comment that this code will be executed also when there are tools?

Reply:
For now at least, unlike OpenAI, there's no notion of mandatory tool to call or not. So my understanding is that the model decides whether to use tools or not on its own without supervision (it's like OpenAI's `auto` mode, I guess.

However, in Gemini, it's also possible to set tools at model definition time, so we don't need to pass tools at each generation request. But soon in the underlying Java SDK, we'll be able to set tools on a per generation request basis, and they will override the tools defined when creating the model.

I guess I'll document the behavior in the JavaDocs.